src/usb.c

```diff
@@ -258,13 +259,99 @@ static int start_rx_loop(struct usb_device *dev)
 	return 0;
 }
 
+static void get_serial_callback(struct libusb_transfer *transfer)
```

Comment:
PS: You'd need to declare this as `static void LIBUSB_CALL get_serial_callback(struct libusb_transfer *transfer)` to make this compile on x86 Windows.

Reply:
I initially had this but I removed it since I noticed the other callback functions in usbmuxd did not. Should I add it back?

Reply:
Good question 😄 . It doesn't hurt to have it on Linux but it's only really needed if you want to compile natively on Windows. Not sure what the repo owners think about it.

Reply:
Can you rebase and change the commit messages to:

```
usb: Store the whole device descriptor in struct usb_device
```
and
```
usb: Use libusb asynchronous I/O for getting initial information

usb_device_add may now be called from libusb main loop via the hotplug callbacks.
No blocking call must occur there and libusb 1.0.21 now returns an error when trying to perform blocking I/O in this callback.
```
(without the before and after stuff)

Then I can merge it.